### PR TITLE
Feature/redirects

### PR DESCRIPTION
**Pull request by yamahito**
Redirect config can be generated using `site:generate-redirects` (which outputs a json file containing all redirect 'stubs'):

```xquery
import module namespace site="http://ns.evolvedbinary.com/sitemap" at "/db/apps/hsg-shell/modules/sitemap-config.xqm";

declare variable $site := doc('/db/apps/hsg-shell/urls.xml')/*;
site:generate-redirects($site) => serialize(map{'method':'json'})
```

This can then be used with the AWS redirect template provided in `resources/cloudfront-redirect` and documented in `cloudfront-redirect/README.md` to enact the redirections on Amazon.  Redirection adds ~1ms to page load times.

### CODE REVIEW

File: cloudfront-redirect/function.js [hunk 1]
```diff
@@ -3,39 +3,57 @@
 const fs = require('fs').promises;
 
 let redirectMap = null;
+const FILE_NAME = '/tmp/redirect-sample.json';
 
 const aws = require('aws-sdk');
 const s3 = new aws.S3({ region: 'us-east-1' });
 const s3Params = {
   Bucket: 'redirectbuket',
```
Comment (plutonik-a) [inline cloudfront-redirect/function.js]: Could this typo interfere at some point, or are both s3Params just placeholders and will be replaced by real values later? (Github wouldn't let me comment on the actual commit, so I leave my comment here...)

**Review by plutonik-a — COMMENTED** [review]
I've gone through all commits related to redirects, tested the redirect test `modules/tests/test-redirects.xqm` (✔︎) and ran the script for checking function `site:generate-redirects($site)` on generating JSON from document data `/db/apps/hsg-shell/urls.xml`, but neither my local installation, nor my dev server was capable of processing this large request.

File: modules/tests/test-pages.xqm [hunk 2]
```diff
@@ -932,15 +932,15 @@ declare %test:assertEquals('true') function x:test-pages-breadcrumb-pocom-role()
                 <a href="/exist/apps/hsg-shell/departmenthistory/people"> People </a>
```
[the remark this answers is not in the document]
Comment (yamahito) [reply modules/tests/test-pages.xqm:932]: I think that this was a bug that @adamretter and I already identified and fixed in XQsuite - the code within the tests do seem to work correctly when run directly.

These tests don't relate to the redirects functionality; we'll try to get them fixed, but I don't think they should hold up this PR.

**Comment by yamahito** [comment]
@joewiz for your information, here is the current JSON redirects file as generated:

```
{
    "/frus150": {
        "status": "301",
        "to": "/departmenthistory/frus-history"
    },
    "/departmenthistory/people/eizenstadt-stuart-c": {
        "status": "301",
        "to": "/departmenthistory/people/eizenstadt-stuart-e"
    },
    "/milestones/1866-1898/Venezuela": {
        "status": "301",
        "to": "/milestones/1866-1898/venezuela"
    },
    "/milestones/1830-1860/Amistad": {
        "status": "301",
        "to": "/milestones/1830-1860/amistad"
    },
    "/milestones/1953-1960/WarsawTreaty": {
        "status": "301",
        "to": "/milestones/1953-1960/warsaw-treaty"
    },
    "/milestones/1953-1960/DienBienPhu": {
        "status": "301",
        "to": "/milestones/1953-1960/dien-bien-phu"
    },
    "/milestones/1921-1936/GreatDepression": {
        "status": "301",
        "to": "/milestones/great-depression"
    },
    "/departmenthistory/people/cook-suzan-johnson": {
        "status": "301",
        "to": "/departmenthistory/people/cook-suzan-denise-johnson"
    },
    "/milestones/1990-2000/foreword": {
        "status": "301",
        "to": "/milestones/1989-1992/foreword"
    },
    "/milestones/1937-1945/LendLease": {
        "status": "301",
        "to": "/milestones/1937-1945/lend-lease"
    },
    "/milestones/1969-1976/VietnamWar": {
        "status": "301",
        "to": "/milestones/1969-1976/vietnam-war"
    },
    "/milestones/1801-1829/LouisianaPurchase": {
        "status": "301",
        "to": "/milestones/1801-1829/louisiana-purchase"
    },
    "/milestones/1953-1960/EisenhowerDoctrine": {
        "status": "301",
        "to": "/milestones/1953-1960/eisenhower-doctrine"
    },
    "/milestones/1969-1976/SALT": {
        "status": "301",
        "to": "/milestones/1969-1976/salt"
    },
    "/departmenthistory/people/rubenstein-daniel-h": {
        "status": "301",
        "to": "/departmenthistory/people/rubinstein-daniel-h"
    },
    "/milestones/1953-1960/Suez": {
        "status": "301",
        "to": "/milestones/1953-1960/suez"
    },
    "/milestones/1921-1936/GoodNeighbor": {
        "status": "301",
        "to": "/milestones/1921-1936/good-neighbor"
    },
    "/departmenthistory/people/clark-dick-clarence": {
        "status": "301",
        "to": "/departmenthistory/people/clark-richard-clarence"
    },
    "/departmenthistory/people/clapp-priscilla-a": {
        "status": "301",
        "to": "/departmenthistory/people/clapp-priscilla-ann"
    },
    "/milestones/1899-1913/RooseveltandMonroeDoctrine": {
        "status": "301",
        "to": "/milestones/1899-1913/roosevelt-and-monroe-doctrine"
    },
    "/departmenthistory/people/pitkin-douglas-c": {
        "status": "301",
        "to": "/departmenthistory/people/pitkin-douglas-alexander"
    },
    "/milestones/1776-1783/ContinentalCongress": {
        "status": "301",
        "to": "/milestones/1776-1783/continental-congress"
    },
    "/milestones/1776-1783/FrenchAlliance": {
        "status": "301",
        "to": "/milestones/1776-1783/french-alliance"
    },
    "/milestones/1961-1968/LimitedBan": {
        "status": "301",
        "to": "/milestones/1961-1968/limited-ban"
    },
    "/departmenthistory/people/anderson-ollie-p-jr": {
        "status": "301",
        "to": "/departmenthistory/people/anderson-ollie-palmer-jr"
    },
    "/milestones/1921-1936/Kellogg": {
        "status": "301",
        "to": "/milestones/1921-1936/kellogg"
    },
    "/departmenthistory/people/gregg-david-l": {
        "status": "301",
        "to": "/departmenthistory/people/gregg-david-lawrence"
    },
    "/milestones/1921-1936/Neutrality_acts": {
        "status": "301",
        "to": "/milestones/1921-1936/neutrality-acts"
    },
    "/milestones/1899-1913/PanamaCanal": {
        "status": "301",
        "to": "/milestones/1899-1913/panama-canal"
    },
    "/milestones/1937-1945/Casablanca": {
        "status": "301",
        "to": "/milestones/1937-1945/casablanca"
    },
    "/milestones/1921-1936/RedLine": {
        "status": "301",
        "to": "/milestones/1921-1936/red-line"
    },
    "/milestones/1914-1920/FourteenPoints": {
        "status": "301",
        "to": "/milestones/1914-1920/fourteen-points"
    },
    "/departmenthistory/people/johnson-lisa-a": {
        "status": "301",
        "to": "/departmenthistory/people/johnson-lisa-anne"
    },
    "/departmenthistory/people/hill-nicholas-m": {
        "status": "301",
        "to": "/departmenthistory/people/hill-nicholas-manning"
    },
    "/departmenthistory/people/smith-keith": {
        "status": "301",
        "to": "/departmenthistory/people/smith-keith-c"
    },
    "/departmenthistory/people/mcmanus-joseph-e": {
        "status": "301",
        "to": "/departmenthistory/people/macmanus-joseph-e"
    },
    "/milestones/1945-1952/KoreanWar2": {
        "status": "301",
        "to": "/milestones/1945-1952/korean-war-2"
    },
    "/milestones/1977-1980/NorthSouthDialogue": {
        "status": "301",
        "to": "/milestones/1977-1980/north-south-dialogue"
    },
    "/milestones/1830-1860/TerritorialExpansion": {
        "status": "301",
        "to": "/milestones/1830-1860/territorial-expansion"
    },
    "/milestones/1921-1936/export-importbank": {
        "status": "301",
        "to": "/milestones/1921-1936/export-import-bank"
    },
    "/departmenthistory/people/futura-toy-julia-ann": {
        "status": "301",
        "to": "/departmenthistory/people/furuta-toy-julia-ann"
    },
    "/milestones/1801-1829/Ruch-Bagot": {
        "status": "301",
        "to": "/milestones/1801-1829/rush-bagot"
    },
    "/departmenthistory/people/gewecke-thomas-h": {
        "status": "301",
        "to": "/departmenthistory/people/gewecke-thomas-howard"
    },
    "/milestones/1953-1960/U2-incident": {
        "status": "301",
        "to": "/milestones/1953-1960/u2-incident"
    },
    "/milestones/1945-1952/MarshallPlan": {
        "status": "301",
        "to": "/milestones/1945-1952/marshall-plan"
    },
    "/departmenthistory/people/wilson-huntington-francis-mairs-huntington-wilson": {
        "status": "301",
        "to": "/departmenthistory/people/wilson-francis-mairs-huntington"
    },
    "/milestones/1801-1829/Florida": {
        "status": "301",
        "to": "/milestones/1801-1829/florida"
    },
    "/milestones/1945-1952/Atomic": {
        "status": "301",
        "to": "/milestones/1945-1952/atomic"
    },
    "/milestones/1953-1960/CENTO": {
        "status": "301",
        "to": "/milestones/1953-1960/cento"
    },
    "/milestones/1921-1936/LondonNavalConf": {
        "status": "301",
        "to": "/milestones/1921-1936/london-naval-conf"
    },
    "/milestones/1866-1898/Hawaii": {
        "status": "301",
        "to": "/milestones/1866-1898/hawaii"
    },
    "/departmenthistory/people/delawie-george-t": {
        "status": "301",
        "to": "/departmenthistory/people/delawie-gregory-t"
    },
    "/milestones/1830-1860/WebsterTreaty": {
        "status": "301",
        "to": "/milestones/1830-1860/webster-treaty"
    },
    "/milestones/1990-2000/Oslo": {
        "status": "301",
        "to": "/milestones/1993-2000/oslo"
    },
    "/milestones/1961-1968/NPT": {
        "status": "301",
        "to": "/milestones/1961-1968/npt"
    },
    "/departmenthistory/people/jones-malika": {
        "status": "301",
        "to": "/departmenthistory/people/james-makila"
    },
    "/milestones/1969-1976/ArabIsraeliWar73": {
        "status": "301",
        "to": "/milestones/1969-1976/arab-israeli-war-1973"
    },
    "/departmenthistory/people/cohen-eliot-a": {
        "status": "301",
        "to": "/departmenthistory/people/cohen-eliot-asher"
    },
    "/milestones/1914-1920/BullittMission": {
        "status": "301",
        "to": "/milestones/1914-1920/bullitt-mission"
    },
    "/milestones/1945-1952/AsiaandAfrica": {
        "status": "301",
        "to": "/milestones/1945-1952/asia-and-africa"
    },
    "/departmenthistory/people/reuben-jeffery": {
        "status": "301",
        "to": "/departmenthistory/people/jeffery-reuben"
    },
    "/milestones/1899-1913/Platt": {
        "status": "301",
        "to": "/milestones/1899-1913/platt"
    },
    "/departmenthistory/people/guthrie-corn-jeri": {
        "status": "301",
        "to": "/departmenthistory/people/guthrie-corn-jeri-schaeffer"
    },
    "/milestones/1953-1960/SEATO": {
        "status": "301",
        "to": "/milestones/1953-1960/seato"
    },
    "/milestones/1945-1952/ArabIsraeliWar": {
        "status": "301",
        "to": "/milestones/1945-1952/arab-israeli-war"
    },
    "/departmenthistory/people/gallegos-gonzalo": {
        "status": "301",
        "to": "/departmenthistory/people/gallegos-gonzalo-rolando"
    },
    "/milestones/1961-1968/TET": {
        "status": "301",
        "to": "/milestones/1961-1968/tet"
    },
    "/milestones/1990-2000": {
        "status": "301",
        "to": "/milestones/1989-1992"
    },
    "/departmenthistory/people/thompson-john-e-w": {
        "status": "301",
        "to": "/departmenthistory/people/thompson-john-edgar-west"
    },
    "/departmenthistory/people/dinkelman-john-w": {
        "status": "301",
        "to": "/departmenthistory/people/dinkelman-john-walter"
    },
    "/departmenthistory/people/moore-jonathan-m": {
        "status": "301",
        "to": "/departmenthistory/people/moore-jonathan-michael"
    },
    "/milestones/1921-1936/USSR": {
        "status": "301",
        "to": "/milestones/1921-1936/ussr"
    },
    "/milestones/1866-1898/Yellow_journalism": {
        "status": "301",
        "to": "/milestones/1866-1898/yellow-journalism"
    },
    "/departmenthistory/people/nolan-edwin-r": {
        "status": "301",
        "to": "/departmenthistory/people/nolan-edwin-richard-jr"
    },
    "/milestones/1861-1865/Confederacy": {
        "status": "301",
        "to": "/milestones/1861-1865/confederacy"
    },
    "/milestones/1899-1913/War": {
        "status": "301",
        "to": "/milestones/1899-1913/war"
    },
    "/milestones/1969-1976/Detente": {
        "status": "301",
        "to": "/milestones/1969-1976/detente"
    },
    "/milestones/1937-1945/YaltaConf": {
        "status": "301",
        "to": "/milestones/1937-1945/yalta-conf"
    },
    "/milestones/1861-1865/TrentAffair": {
        "status": "301",
        "to": "/milestones/1861-1865/trent-affair"
    },
    "/milestones/1866-1898/AlaskaPurchase": {
        "status": "301",
        "to": "/milestones/1866-1898/alaska-purchase"
    },
    "/milestones/1914-1920/WWI": {
        "status": "301",
        "to": "/milestones/1914-1920/wwi"
    },
    "/milestones/1945-1952/CreationIsrael": {
        "status": "301",
        "to": "/milestones/1945-1952/creation-israel"
    },
    "/milestones/1945-1952/Kennan": {
        "status": "301",
        "to": "/milestones/1945-1952/kennan"
    },
    "/departmenthistory/people/james-malika": {
        "status": "301",
        "to": "/departmenthistory/people/james-makila"
    },
    "/departmenthistory/people/hachigan-nina": {
        "status": "301",
        "to": "/departmenthistory/people/hachigian-nina"
    },
    "/milestones/1776-1783/ModelTreaty": {
        "status": "301",
        "to": "/milestones/1776-1783/model-treaty"
    },
    "/milestones/1961-1968/ArabIsraeliWar67": {
        "status": "301",
        "to": "/milestones/1961-1968/arab-israeli-war-1967"
    },
    "/milestones/1969-1976/SouthAsia": {
        "status": "301",
        "to": "/milestones/1969-1976/south-asia"
    },
    "/milestones/1784-1800/XYZ": {
        "status": "301",
        "to": "/milestones/1784-1800/xyz"
    },
    "/milestones/1945-1952/ANZUS": {
        "status": "301",
        "to": "/milestones/1945-1952/anzus"
    },
    "/milestones/1830-1860/OregonTerritory": {
        "status": "301",
        "to": "/milestones/1830-1860/oregon-territory"
    },
    "/milestones/1953-1960/Sputnik": {
        "status": "301",
        "to": "/milestones/1953-1960/sputnik"
    },
    "/departmenthistory/people/de-pree-william-ames": {
        "status": "301",
        "to": "/departmenthistory/people/de-pree-willard-ames"
    },
    "/departmenthistory/people/brown-natalie": {
        "status": "301",
        "to": "/departmenthistory/people/brown-natalie-e"
    },
    "/milestones/1981-1989/FallofCommunism": {
        "status": "301",
        "to": "/milestones/1989-1992/fall-of-communism"
    },
    "/departmenthistory/people/barnes-douglas-m": {
        "status": "301",
        "to": "/departmenthistory/people/barnes-douglas-malcolm"
    },
    "/milestones/1945-1952/BerlinCrises": {
        "status": "301",
        "to": "/milestones/1945-1952/berlin-airlift"
    },
    "/departmenthistory/people/korn-david-a": {
        "status": "301",
        "to": "/departmenthistory/people/korn-david-adolph"
    },
    "/milestones/1937-1945/WarTimeConferences": {
        "status": "301",
        "to": "/milestones/1937-1945/war-time-conferences"
    },
    "/milestones/1866-1898/Blaine": {
        "status": "301",
        "to": "/milestones/1866-1898/blaine"
    },
    "/milestones/1937-1945/BrettonWoods": {
        "status": "301",
        "to": "/milestones/bretton-woods"
    },
    "/departmenthistory/people/butcher-duane-c": {
        "status": "301",
        "to": "/departmenthistory/people/butcher-duane-clemens"
    },
    "/milestones/1899-1913/ChineseRev": {
        "status": "301",
        "to": "/milestones/1899-1913/chinese-rev"
    },
    "/milestones/1776-1783/SecretCommittee": {
        "status": "301",
        "to": "/milestones/1776-1783/secret-committee"
    },
    "/milestones/1914-1920/League": {
        "status": "301",
        "to": "/milestones/1914-1920/league"
    },
    "/milestones/1953-1960/BerlinCrises": {
        "status": "301",
        "to": "/milestones/1953-1960/berlin-crises"
    },
    "/departmenthistory/people/brown-david-e": {
        "status": "301",
        "to": "/departmenthistory/people/brown-david-edward"
    },
    "/milestones/1866-1898/Spanish_american_war": {
        "status": "301",
        "to": "/milestones/1866-1898/spanish-american-war"
    },
    "/milestones/1937-1945/ChineseExclusionActRepeal": {
        "status": "301",
        "to": "/milestones/1937-1945/chinese-exclusion-act-repeal"
    },
    "/milestones/1937-1945/Internationalism": {
        "status": "301",
        "to": "/milestones/1937-1945/internationalism"
    },
    "/milestones/1801-1829/Warof1812": {
        "status": "301",
        "to": "/milestones/1801-1829/war-of-1812"
    },
    "/milestones/1784-1800/ConventionandRatification": {
        "status": "301",
        "to": "/milestones/1784-1800/convention-and-ratification"
    },
    "/departmenthistory/people/verveer-mellane": {
        "status": "301",
        "to": "/departmenthistory/people/verveer-melanne"
    },
    "/departmenthistory/people/whidden-benjamin-f": {
        "status": "301",
        "to": "/departmenthistory/people/whidden-benjamin-franklin"
    },
    "/departmenthistory/people/brush-jennifer-l": {
        "status": "301",
        "to": "/departmenthistory/people/brush-jennifer-leigh"
    },
    "/milestones/1953-1960/ChinaTalks": {
        "status": "301",
        "to": "/milestones/1953-1960/china-talks"
    },
    "/milestones/1937-1945/UN": {
        "status": "301",
        "to": "/milestones/1937-1945/un"
    },
    "/milestones/1969-1976/Helsinki": {
        "status": "301",
        "to": "/milestones/1969-1976/helsinki"
    },
    "/milestones/1776-1783/Declaration": {
        "status": "301",
        "to": "/milestones/1776-1783/declaration"
    },
    "/departmenthistory/people/gilmore-harry-g": {
        "status": "301",
        "to": "/departmenthistory/people/gilmore-harry-joseph"
    },
    "/departmenthistory/people/bremner-susan": {
        "status": "301",
        "to": "/departmenthistory/people/bremner-susan-lenore"
    },
    "/milestones/1784-1800/FrenchRev": {
        "status": "301",
        "to": "/milestones/1784-1800/french-rev"
    },
    "/milestones/1945-1952/KoreanWar": {
        "status": "301",
        "to": "/milestones/1945-1952/korean-war"
    },
    "/departmenthistory/people/saltzman-charles-eckridge": {
        "status": "301",
        "to": "/departmenthistory/people/saltzman-charles-eskridge"
    },
    "/milestones/1953-1960/khrushchev-20thc-congress": {
        "status": "301",
        "to": "/milestones/1953-1960/khrushchev-20th-congress"
    },
    "/milestones/1977-1980/Olympic": {
        "status": "301",
        "to": "/milestones/1977-1980/olympic"
    },
    "/milestones/1784-1800/PickneyTreaty": {
        "status": "301",
        "to": "/milestones/1784-1800/pickney-treaty"
    },
    "/milestones/1899-1913/DollarDiplo": {
        "status": "301",
        "to": "/milestones/1899-1913/dollar-diplo"
    },
    "/milestones/1801-1829/BarbaryWars": {
        "status": "301",
        "to": "/milestones/1801-1829/barbary-wars"
    },
    "/departmenthistory/people/thompson-dean-r": {
        "status": "301",
        "to": "/departmenthistory/people/thompson-dean-richard"
    },
    "/departmenthistory/people/tracey-lynne-m": {
        "status": "301",
        "to": "/departmenthistory/people/tracy-lynne-m"
    },
    "/milestones/1784-1800/JayTreaty": {
        "status": "301",
        "to": "/milestones/1784-1800/jay-treaty"
    },
    "/departmenthistory/people/norris-chester-e": {
        "status": "301",
        "to": "/departmenthistory/people/norris-chester-edward-jr"
    },
    "/departmenthistory/people/kirk-robert-c": {
        "status": "301",
        "to": "/departmenthistory/people/kirk-robert-crothers"
    },
    "/milestones/1961-1968/AllianceforProgress": {
        "status": "301",
        "to": "/milestones/1961-1968/alliance-for-progress"
    },
    "/milestones/1866-1898/Mahan": {
        "status": "301",
        "to": "/milestones/1866-1898/mahan"
    },
    "/milestones/1977-1980/TrilateralDiplomacy": {
        "status": "301",
        "to": "/milestones/1977-1980/trilateral-diplomacy"
    },
    "/milestones/1914-1920/Haiti": {
        "status": "301",
        "to": "/milestones/1914-1920/haiti"
    },
    "/milestones/1937-1945/PotsdamConf": {
        "status": "301",
        "to": "/milestones/1937-1945/potsdam-conf"
    },
    "/departmenthistory/people/hankins-dennis": {
        "status": "301",
        "to": "/departmenthistory/people/hankins-dennis-b"
    },
    "/milestones/1945-1952/NATO": {
        "status": "301",
        "to": "/milestones/1945-1952/nato"
    },
    "/milestones/1961-1968/GulfofTonkin": {
        "status": "301",
        "to": "/milestones/1961-1968/gulf-of-tonkin"
    },
    "/milestones/1830-1860/OpeningtoJapan": {
        "status": "301",
        "to": "/milestones/1830-1860/opening-to-japan"
    },
    "/milestones/1937-1945/TehranConf": {
        "status": "301",
        "to": "/milestones/1937-1945/tehran-conf"
    },
    "/departmenthistory/people/bernicat-marcia-s-b": {
        "status": "301",
        "to": "/departmenthistory/people/bernicat-marcia-stephens-bloom"
    },
    "/milestones/1830-1860/GadsdenPurchase": {
        "status": "301",
        "to": "/milestones/1830-1860/gadsden-purchase"
    },
    "/milestones/1899-1913/PortsmouthTreaty": {
        "status": "301",
        "to": "/milestones/1899-1913/portsmouth-treaty"
    },
    "/milestones/1750-1775/ParliamentaryTaxation": {
        "status": "301",
        "to": "/milestones/1750-1775/parliamentary-taxation"
    },
    "/departmenthistory/people/jones-paul-w": {
        "status": "301",
        "to": "/departmenthistory/people/jones-paul-wayne"
    },
    "/milestones/1969-1976/OPEC": {
        "status": "301",
        "to": "/milestones/1969-1976/oil-embargo"
    },
    "/milestones/1921-1936/Mukden_incident": {
        "status": "301",
        "to": "/milestones/1921-1936/mukden-incident"
    },
    "/departmenthistory/people/ofriel-paul": {
        "status": "301",
        "to": "/departmenthistory/people/ofriel-paul-christopher"
    },
    "/milestones/1990-2000/StrategicArmsReduction": {
        "status": "301",
        "to": "/milestones/1989-1992/strategic-arms-reduction"
    },
    "/milestones/1953-1960/TaiwanSTraitCrises": {
        "status": "301",
        "to": "/milestones/1953-1960/taiwan-strait-crises"
    },
    "/milestones/1750-1775/FrenchIndianWar": {
        "status": "301",
        "to": "/milestones/1750-1775/french-indian-war"
    },
    "/milestones/1750-1775/AlbanyPlan": {
        "status": "301",
        "to": "/milestones/1750-1775/albany-plan"
    },
    "/milestones/1830-1860/TexasAnnexation": {
        "status": "301",
        "to": "/milestones/1830-1860/texas-annexation"
    },
    "/milestones/1945-1952/BaruchPlans": {
        "status": "301",
        "to": "/milestones/1945-1952/baruch-plans"
    },
    "/milestones/1945-1952/ChineseRev": {
        "status": "301",
        "to": "/milestones/1945-1952/chinese-rev"
    },
    "/milestones/1776-1783/BFranklin": {
        "status": "301",
        "to": "/milestones/1776-1783/b-franklin"
    },
    "/milestones/1921-1936/ImmigrationAct": {
        "status": "301",
        "to": "/milestones/1921-1936/immigration-act"
    },
    "/milestones/1784-1800/HaitianRev": {
        "status": "301",
        "to": "/milestones/1784-1800/haitian-rev"
    },
    "/departmenthistory/people/bellochi-natale-h": {
        "status": "301",
        "to": "/departmenthistory/people/bellocchi-natale-h"
    },
    "/milestones/1899-1913/JapaneseRelations": {
        "status": "301",
        "to": "/milestones/1899-1913/japanese-relations"
    },
    "/milestones/1961-1968/CubanMissile": {
        "status": "301",
        "to": "/milestones/1961-1968/cuban-missile-crisis"
    },
    "/milestones/1750-1775/Incidents": {
        "status": "301",
        "to": "/milestones/1750-1775/incidents"
    },
    "/departmenthistory/people/jarimillo-mari-luci": {
        "status": "301",
        "to": "/departmenthistory/people/jaramillo-mari-luci"
    },
    "/departmenthistory/people/fox-carl-b": {
        "status": "301",
        "to": "/departmenthistory/people/fox-carl-benjamin"
    },
    "/milestones/1921-1936/Geneva": {
        "status": "301",
        "to": "/milestones/1921-1936/geneva"
    },
    "/departmenthistory/people/carlisle-samuel-s": {
        "status": "301",
        "to": "/departmenthistory/people/carlisle-samuel-stewart"
    },
    "/milestones/1866-1898/CommodoreShufeldt": {
        "status": "301",
        "to": "/milestones/1866-1898/commodore-shufeldt"
    },
    "/departmenthistory/people/ferrill-arlene": {
        "status": "301",
        "to": "/departmenthistory/people/ferrill-arlene-lorraine"
    },
    "/milestones/1921-1936/Dawes": {
        "status": "301",
        "to": "/milestones/1921-1936/dawes"
    },
    "/milestones/1914-1920/ParisPeace": {
        "status": "301",
        "to": "/milestones/1914-1920/paris-peace"
    },
    "/milestones/1776-1783/Treaty": {
        "status": "301",
        "to": "/milestones/1776-1783/treaty"
    },
    "/milestones/1937-1945/MexicanOil": {
        "status": "301",
        "to": "/milestones/1937-1945/mexican-oil"
    },
    "/milestones/1969-1976/Angola": {
        "status": "301",
        "to": "/milestones/1969-1976/angola"
    },
    "/milestones/1861-1865/Blockade": {
        "status": "301",
        "to": "/milestones/1861-1865/blockade"
    },
    "/milestones/1861-1865/Victory": {
        "status": "301",
        "to": "/milestones/1861-1865/victory"
    },
    "/milestones/1899-1913/HayandChina": {
        "status": "301",
        "to": "/milestones/1899-1913/hay-and-china"
    },
    "/milestones/1937-1945/AtlanticConf": {
        "status": "301",
        "to": "/milestones/1937-1945/atlantic-conf"
    },
    "/departmenthistory/people/jacob-charles-d": {
        "status": "301",
        "to": "/departmenthistory/people/jacob-charles-donald"
    },
    "/milestones/1921-1936/NavalConference": {
        "status": "301",
        "to": "/milestones/1921-1936/naval-conference"
    },
    "/milestones/1801-1829/Monroe": {
        "status": "301",
        "to": "/milestones/1801-1829/monroe"
    },
    "/departmenthistory/people/schwartz-stephen-m": {
        "status": "301",
        "to": "/departmenthistory/people/schwartz-stephen-michael"
    },
    "/milestones/1937-1945/USSoviet": {
        "status": "301",
        "to": "/milestones/1937-1945/us-soviet"
    },
    "/milestones/1921-1936/Protectionism": {
        "status": "301",
        "to": "/milestones/1921-1936/protectionism"
    },
    "/milestones/1830-1860/Liberia": {
        "status": "301",
        "to": "/milestones/1830-1860/liberia"
    },
    "/departmenthistory/people/fitrell-troy-d": {
        "status": "301",
        "to": "/departmenthistory/people/fitrell-troy-damian"
    },
    "/milestones/1945-1952/JapanReconstruction": {
        "status": "301",
        "to": "/milestones/1945-1952/japan-reconstruction"
    },
    "/milestones/1801-1829/NapoleonicWars": {
        "status": "301",
        "to": "/milestones/1801-1829/napoleonic-wars"
    },
    "/milestones/1784-1800/Loans": {
        "status": "301",
        "to": "/milestones/1784-1800/loans"
    },
    "/departmenthistory/people/dwyer-stuart": {
        "status": "301",
        "to": "/departmenthistory/people/dwyer-stuart-anderson"
    },
    "/departmenthistory/people/daley-thomas-e": {
        "status": "301",
        "to": "/departmenthistory/people/daley-thomas-edward"
    },
    "/milestones/1784-1800/WashingtonFarewell": {
        "status": "301",
        "to": "/milestones/1784-1800/washington-farewell"
    },
    "/milestones/1776-1783/Articles": {
        "status": "301",
        "to": "/milestones/1776-1783/articles"
    },
    "/milestones/1830-1860/China2": {
        "status": "301",
        "to": "/milestones/1830-1860/china-2"
    },
    "/milestones/1866-1898/Telegraph": {
        "status": "301",
        "to": "/milestones/1866-1898/telegraph"
    },
    "/milestones/1937-1945/AmericanIsolationism": {
        "status": "301",
        "to": "/milestones/1937-1945/american-isolationism"
    },
    "/milestones/1861-1865/FrenchIntervention": {
        "status": "301",
        "to": "/milestones/1861-1865/french-intervention"
    },
    "/milestones/1861-1865/Alabama": {
        "status": "301",
        "to": "/milestones/1861-1865/alabama"
    },
    "/milestones/1830-1860/China1": {
        "status": "301",
        "to": "/milestones/1830-1860/china-1"
    },
    "/departmenthistory/people/watson-samuel-r": {
        "status": "301",
        "to": "/departmenthistory/people/watson-samuel-robert"
    },
    "/milestones/1866-1898/ChineseImmigration": {
        "status": "301",
        "to": "/milestones/1866-1898/chinese-immigration"
    },
    "/milestones/1945-1952/ImmigrationAct": {
        "status": "301",
        "to": "/milestones/1945-1952/immigration-act"
    },
    "/milestones/1953-1960/BandungConf": {
        "status": "301",
        "to": "/milestones/1953-1960/bandung-conf"
    },
    "/milestones/1784-1800/CitizenGenet": {
        "status": "301",
        "to": "/milestones/1784-1800/citizen-genet"
    },
    "/departmenthistory/people/hale-david": {
        "status": "301",
        "to": "/departmenthistory/people/hale-david-m"
    },
    "/milestones/1866-1898/Burlingame-SewardTreaty": {
        "status": "301",
        "to": "/milestones/1866-1898/burlingame-seward-treaty"
    },
    "/milestones/1945-1952/TrumanDoctrine": {
        "status": "301",
        "to": "/milestones/1945-1952/truman-doctrine"
    },
    "/departmenthistory/people/bivins-miles-t": {
        "status": "301",
        "to": "/departmenthistory/people/bivins-miles-teel"
    },
    "/milestones/1830-1860/IndianTreaties": {
        "status": "301",
        "to": "/milestones/1830-1860/indian-treaties"
    },
    "/departmenthistory/people/goldrich-ethan-a": {
        "status": "301",
        "to": "/departmenthistory/people/goldrich-ethan-aaron"
    },
    "/milestones/1750-1775/TreatyofParis": {
        "status": "301",
        "to": "/milestones/1750-1775/treaty-of-paris"
    },
    "/milestones/1981-1989/APEC": {
        "status": "301",
        "to": "/milestones/1989-1992/apec"
    },
    "/milestones/1945-1952/NationalSecurityAct": {
        "status": "301",
        "to": "/milestones/1945-1952/national-security-act"
    },
    "/milestones/1961-1968/IndiaPakistanWar": {
        "status": "301",
        "to": "/milestones/1961-1968/india-pakistan-war"
    },
    "/milestones/1830-1860/PacificExpansion": {
        "status": "301",
        "to": "/milestones/1830-1860/pacific-expansion"
    },
    "/milestones/1990-2000/Apartheid": {
        "status": "301",
        "to": "/milestones/1989-1992/apartheid"
    },
    "/milestones/1750-1775/ProclamationLine1763": {
        "status": "301",
        "to": "/milestones/1750-1775/proclamation-line"
    }
}
```

File: cloudfront-redirect/function.js (hunk 1 above)
Comment (yamahito) [reply cloudfront-redirect/function.js]: I believe they are just placeholders.

**Review by joewiz — APPROVED** [review]
The list of redirects looks good, as does the PR as a whole.

**Pull request merged**